### ISSUE TITLE
Implement migration "up to" and "down to" specific migration file

### ISSUE THREAD

**Issue by gpetrov** [knex/knex#4260]
# Feature request

When running migration bound to version control, you want to sequentially go up to specific version and also sequentially go down to other specific version on revert. This is how "up to" and "down to" migration commands  are implemented in other migration solutions like: https://github.com/tj/node-migrate#running-migrations

The migrations need always to be executed sequential from the last active to (and including) the specified "to" - also all in a single batch.

Please implement this as knex.migrate.upTo and knex.migrate.downTo api calls or just allow to pass "to" as extra configuration options next to name in the regular up and down methods.

**Comment by elhigu** [comment]
Just as a side not. Currently actually migrations are not necessary ran in the same order to the database in which they are in directory listing (is new file is added to middle of the list, it will be applied to top of all migrations; that might happen for example when two feature branches are merged to main and the feature which was created later is merged first), so it should be also decided how knex would work in that corner case.

**Comment by gpetrov** [comment]
Good point but as all files are prefixed with timestamp a simple sort order should bring them in the right order before processing.

The tricky part is when using version control for publishing new releases of your code and migrations and when you want to be able to revert and bring down the necessary migrations.

So when upgrading to new version from git, you will first push the new code then apply migrations.

But if you decide to revert you have to bring down the last migrations first and then revert the code to the previous  version tag.

So that is a bit tricky and not really atomic...

**Comment by jcwatson11** [comment]
Hello and thank you for posting this feature request. I find it is one that I can relate to. I've been a release manager for the past 10 years. This feature is something that we have solved in other frameworks and would be really nice to have in this one.

Here's our solution in principle:

Given Philosophy:

1. From a software release perspective, rollbacks happen when a release fails and we want to restore a deployment target to its previous state.
2. Software releases generally include several migrations that are being released as a batch along with code that makes use of the new schema/data included in the migrations.
3. The migrations table has a batch number that groups these batches together in any case when we run ```knex migrate:latest```
4. If a given release with a group of migrations ran successfully, we would want to rollback the code and the migrations related to that code at the same time.
5. The release process should not have to worry about implementation details for the migration tool.

Current Situation:
Presently, achieving success here is possible only by evaluating the entire list of migrations related to a given batch number and rolling those migrations back individually in reverse order. This requires a release manager to violate number 5 and write a program. It's not really a big deal. But I believe the following conclusions are still true:

Therefore:
1. Associating a software release version to the batch number is key to being able to rollback to the appropriate place from a release perspective. This is the release manager's responsibility, and knex shouldn't care what that association is. But the importance of being able to rollback a batch as a whole is key.
2. It should be well within reason to add a knex native command that does something like ```knex migrate:rollback --batch <BATCHNUMBER>```... which would be expected to rollback all migrations in sequence later than and including <BATCHNUMBER> in reverse order.
3. This would allow release managers like me to be responsible only for associating our software release with the batch number and automating any release commands... as opposed to also creating a more detailed program that does number 5 above.

Thanks for reading!

**Comment by pinko-fowle** [comment]
We've used [knex-migrate](https://www.npmjs.com/package/knex-migrate) package to do this upgrade/rollback to specific versions.

**Comment by mercmobily** [comment]
I am closing this in favour of #6232 which is an official feature request. Please write there if you have any comments!
